event/channel_bus.go

```diff
@@ -94,18 +106,22 @@ func (b *Bus) Subscribe(events ...Name) (*Subscription, error) {
 // Unsubscribe removes all event subscriptions and closes the subscription channel.
 //
 // Will do nothing if this object is already closed.
-func (b *Bus) Unsubscribe(sub *Subscription) {
+func (b *channelBus) Unsubscribe(sub Subscription) {
 	b.closeMutex.RLock()
 	defer b.closeMutex.RUnlock()
 
 	if b.isClosed {
 		return
 	}
-	b.commandChannel <- unsubscribeCommand(sub)
+	s, ok := sub.(*channelSub)
+	if !ok {
+		return
```

Comment:
thought: Did you you consider panicking here instead? I worry that this might cause hidden bugs long term, with a silent return.

suggestion: I think a comment here either warning that this is not something we want long term (maybe linked to a new issue), or, if you do want this long term, a description of why you think it is a good solution as otherwise it looks a bit broken.